### ISSUE TITLE
[Question] Raylet may not be launched alongside each executor?

### ISSUE THREAD

**Issue by hkvision** [intel-analytics/analytics-zoo#3721]
https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/zoo/ray/raycontext.py#L504

```
ray_rdd = self.sc.range(0, self.num_ray_nodes,
                                numSlices=self.num_ray_nodes)

process_infos = ray_rdd.barrier().mapPartitions(
    self.ray_service.gen_ray_start(self.cluster_ips[0])).collect()
```
There is no guarantee that each task is combined with each executor? (especially when not all the executors have been already launched in the cluster?

cc @jason-dai

**Comment by hkvision** [comment]
Setting spark.task.cpus=executor cores could guarantee each executor has one task to get the ip? @jason-dai

**Comment by jason-dai** [comment]
> Setting spark.task.cpus=executor cores could guarantee each executor has one task to get the ip? @jason-dai

Yes, but that's a global setting that applies to all Spark jobs.

**Comment by hkvision** [comment]
> > Setting spark.task.cpus=executor cores could guarantee each executor has one task to get the ip? @jason-dai
> 
> Yes, but that's a global setting that applies to all Spark jobs.

Yeah, so besides this, it cannot be guaranteed.

**Comment by hkvision** [comment]
Launching total core number of tasks (instead of executor number of tasks) may make each executor have a task, but seems there is no obvious way to get the executor id of the task to make each executor launch ray only once.
